lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -353,8 +393,16 @@ let stage2 dis stage1 =
             Dis.stop s (Dis.insns s)) |>
       List.map ~f:(function
           | mem, None -> mem,(None,None)
-          | mem, (Some ins as insn) -> match stage1.lift mem ins with
-            | Ok bil -> mem,(insn,Some bil)
+          | mem, (Some ins as insn) ->
+            match stage1.lift mem ins with
+            | Ok bil ->
+              let bil =
+                if has_jump bil then
```

Comment:
it should be required for an instruction to have a jump, so we should remove this check and ensure that the BIL code always contains the sum of the destinations returned by the lifter and by the brancher.